etcdserver/api/v3rpc/watch.go

```diff
@@ -140,7 +140,11 @@ func (ws *watchServer) Watch(stream pb.Watch_WatchServer) (err error) {
 	// deadlock when calling sws.close().
 	go func() {
 		if rerr := sws.recvLoop(); rerr != nil {
-			plog.Debugf("failed to receive watch request from gRPC stream (%q)", rerr.Error())
+			if stream.Context().Err() == context.Canceled {
```

Comment:
both context.Cancel and context.DeadlineExceeded should be ignored i think.

Reply:
Also we should have a shared function for all streaming API logs https://github.com/coreos/etcd/pull/8939 (e.g. lease keepalive).

Reply:
@gyuho thanks! I just learned how to use lease :-) and could see the related log messages. So also modifying the message on LOC 110 (for user cancelled lease) in #8939 is what your suggesting right? I know you suggested to refer to the ```toErr``` function in ```etcd/clientv3/client.go``` in the issue discussion but I believe that was just to learn about different error code and not to make any changes there. Thanks!

Reply:
Yes, just like we do in clientv3, there should be an error parsing function on this `rerr` error. Convert it to gRPC `*status.statusError` and match the error code. If the error is canceled or deadline exceeded, we log with debug level.